### ISSUE TITLE
Bundle task with the extension

### ISSUE THREAD

**Issue by jzaefferer** [Microsoft/vscode-eslint#247]
If I understand this comment correctly ( https://github.com/Microsoft/vscode/issues/1435#issuecomment-303058258 ), the latest VS Code supports extension bundling tasks, which sounds like a good (better?) solution to defining tasks globally.

In other words, it might now be possible to replace the instructions here ( https://github.com/Microsoft/vscode-eslint/tree/master/eslint#using-the-extension-with-vs-codes-task-running ) with "Run the bundled ESLint task".

**Comment by marvinhagemeister** [comment]
This portion of [tslint](https://github.com/Microsoft/vscode-tslint) might be really interesting. They define a `ProblemMatcher` in `package.json` that runs automatically. Meaning all errors of all files show up in the Problems view. No need to create custom tasks for linting.

https://github.com/Microsoft/vscode-tslint/blob/master/tslint/package.json#L157

**Comment by dbaeumer** [comment]
Yes, that is the goal. But this is not the default yet since we want to get more feedback on this from the community. But users can opt into it using a dummy tasks.json with the content 
```
{
    "version": "2.0.0"
}
```

**Comment by nickyoung23** [comment]
I'm trying to test the bundling of tasks in a extension (using Code Insiders), but can't get it to work. My understanding is that there will be a "tasks" key within "contributes" in package.json, is that correct?

I have assumed that this will be an array in the same way that "languages" and "grammars" are, but have tried many different options without success. Please could you provide a quick example to get me started.

**Comment by dbaeumer** [comment]
@nickyoung23 task will be contributed by code, not descriptively via the package.json. This gives more flexibility in terms of customizing tasks. The API to do so is still in proposed state and available here: https://github.com/Microsoft/vscode/blob/master/src/vs/vscode.proposed.d.ts

**Comment by Hirse** [comment]
What is the state of this?
Does the task contributed by the extension use the latest version yet?
